src/main/java/echoserver/Listenable.java

```diff
@@ -0,0 +1,8 @@
+package echoserver;
+
+import java.io.IOException;
+
+public interface Listenable {
+    Connection open() throws IOException;
```

Comment:
Usually, when I describe a roles as `able`, it will correspond to messages that the object will receive. 

For instance, a `Callable` would have a `call` method.

Thus, I might go with renaming `open` to `listen` or renaming the Interface to an `Openable` or something.  Just a suggestion though. No change needed.

Reply:
Okay will make the change